bin/desi_assemble_tilepix

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+"""
+Assemble individual tilepix files into output summary files mapping which
+tiles cover which healpix
+"""
+
+import os, sys, glob, json
+from copy import deepcopy
+import argparse
+
+def getpix(tilepix, tileids):
+    """
+    Return which healpix are covered by these tileids
+
+    Args:
+        tilepix: dict[tileid][petal] = list of healpix
+        tileids: array-like list of tileids
+
+    Returns:
+        array of healpix covered by these tiles according to tilepix
+    """
+    pix = list()
+    for tileid in tileids:
+        t = str(tileid)
+        for petal in range(10):
+            p = str(petal)
+            if p in tilepix[t]:
+                pix.extend(tilepix[t][p])
+
+    pix = np.unique(pix)
+    return pix
+
+#-------------------------------------------------------------------------
+
+p = argparse.ArgumentParser()
+p.add_argument('-s', '--specprod', type=str,
+        help='override $SPECPROD')
+p.add_argument('-o', '--outfile', type=str,
+        help='output fits file; .json will also be created')
+p.add_argument('-e', '--expfile', type=str,
+        help='input exposures fits file (default exposures-$SPECPROD.fits)')
+p.add_argument('--check', action='store_true',
+        help='perform checks comparing tilepix map to files on disk')
+
+args = p.parse_args()
+
+#- not just asking for --help, so proceed with slower imports
+import numpy as np
+from astropy.table import Table
+
+from desiutil.log import get_logger
+import desispec.io
+from desispec.io.util import parse_cameras, camword_union
+
+log = get_logger()
+
+#- defaults for args
+if args.specprod is not None:
+    os.environ['SPECPROD'] = args.specprod
+
+specprod = os.environ['SPECPROD']
+
+if args.outfile is not None:
+    args.outfile = os.path.abspath(args.outfile)
+
+reduxdir = desispec.io.specprod_root()
+log.info(f'Inspecting {reduxdir}')
+
+if args.expfile is None:
+    args.expfile = f'{reduxdir}/exposures-{specprod}.fits'
+
+#- read exposures and frames in this production
+log.info(f'Reading exposures from {args.expfile}')
+exp = Table.read(args.expfile, 'EXPOSURES')
+frames = Table.read(args.expfile, 'FRAMES')
+
+#- Load all individual tilepix files
+tileids, firstindex = np.unique(exp['TILEID'], return_index=True) 
+tilepix = dict()
+for tileid, i in zip(tileids, firstindex):
+    night = exp['NIGHT'][i]
+    expid = exp['EXPID'][i]
+    fn = desispec.io.findfile('tilepix', tile=tileid, night=night, expid=expid)
+    with open(fn) as fx:
+        tilepix.update( json.load(fx) )
+
+#- filter out petals that don't have frames
+for tileid in tileids:
+    ii = frames['TILEID'] == tileid
+    camword = parse_cameras(np.unique(frames['CAMERA'][ii]), loglevel='warning')
+    #- should only be complete petals, not individual cameras left
+    assert 'b' not in camword
+    assert 'r' not in camword
+    assert 'z' not in camword
+    for petal in range(10):
+        if camword_union([camword, f'a{petal}']) != camword:
```

Comment:
Given the above assertions I think you can just check `if str(petal) not in camword`, couldn't you?

Reply:
good idea, that is cleaner.  I switched to that.